ssg/build_ovals.py

```diff
@@ -250,6 +254,17 @@ def _check_oval_version_from_oval(xml_content, oval_version):
         return True
 
 
+def _check_rule_id(oval_file_tree, rule_id, file_path):
+    for definition in oval_file_tree.findall(
+            "./{%s}def-group/{%s}definition" % (oval_ns, oval_ns)):
+        definition_id = definition.get("id")
+        if definition_id != rule_id:
+            msg = (
+                "OVAL definition ID '%s' doesn't match rule ID '%s' "
+                "in '%s'\n" % (definition_id, rule_id, file_path))
+            sys.stderr.write(msg)
```

Comment:
It would be nice to have the issue discovery and issue handling separate, preferably in a separate function.

Reply:
I would prefer to keep it here because I find the information in this message valuable.

Reply:
I am not asking to drop anything, just to decouple the detection from reporting. Right now, the function is immobile - printing that exact message may not be desirable side-effect all of the time.